src/cpp/CMakeLists.txt

```diff
@@ -355,6 +361,110 @@ else()
+
+# database library includes
+if (UNIX)
+   if (APPLE)
+      # We assume that libpq is available under the default brew install location
+      # since we install postgresql via brew in install-dependencies-osx
+      include_directories(SYSTEM "/usr/local/Cellar/postgresql/12.1/include")
+      include_directories(SYSTEM "/usr/local/Cellar/postgresql/12.1/include/libpq")
+   else()
+      include_directories(SYSTEM "/usr/include/postgresql")
+   endif()
+elseif(WIN32)
+   include_directories(SYSTEM "${RSTUDIO_WINDOWS_DEPENDENCIES_DIR}/install-soci/sqlite/sqlite-amalgamation-3310100")
+   include_directories(SYSTEM "${RSTUDIO_WINDOWS_DEPENDENCIES_DIR}/install-soci/postgresql/include")
+endif()
+
+# find SOCI libraries
+if(UNIX)
+   set(SOCI_LIBRARY_DIR "${RSTUDIO_TOOLS_SOCI}/build/lib")
+   if (NOT APPLE)
+      set(LIB_SUFFIX ".so")
+      set(SOCI_LIB_SYMLINK "${LIB_SUFFIX}.4.0")
+      set(SOCI_LIB_SUFFIX "${LIB_SUFFIX}.4.0.0")
+   else()
+      set(LIB_SUFFIX ".dylib")
+      set(SOCI_LIB_SYMLINK ".4.0${LIB_SUFFIX}")
+      set(SOCI_LIB_SUFFIX ".4.0.0${LIB_SUFFIX}")
+   endif()
+   file(GLOB_RECURSE SOCI_LIBRARIES "${SOCI_LIBRARY_DIR}/*${LIB_SUFFIX}")
+else()
+   set(SOCI_LIBRARY_DIR "${RSTUDIO_TOOLS_SOCI}/build/${SOCI_ARCH}/lib")
+   if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+       file(GLOB_RECURSE SOCI_LIBRARIES "${SOCI_LIBRARY_DIR}/Debug/*.lib")
+       list(APPEND SOCI_LIBRARIES "${WIN_LDAP_LIBRARY}")
+       list(APPEND SOCI_LIBRARIES "${WIN_SSPI_LIBRARY}")
+       list(APPEND SOCI_LIBRARIES "${RSTUDIO_WINDOWS_DEPENDENCIES_DIR}/install-soci/sqlite/sqlite3-debug-${SOCI_ARCH}.lib")
+       list(APPEND SOCI_LIBRARIES "${RSTUDIO_WINDOWS_DEPENDENCIES_DIR}/install-soci/postgresql/lib/libpq-debug-${SOCI_ARCH}.lib")
+       list(APPEND SOCI_LIBRARIES "${RSTUDIO_WINDOWS_DEPENDENCIES_DIR}/install-soci/postgresql/lib/libpgcommon-debug-${SOCI_ARCH}.lib")
+       list(APPEND SOCI_LIBRARIES "${RSTUDIO_WINDOWS_DEPENDENCIES_DIR}/install-soci/postgresql/lib/libpgport-debug-${SOCI_ARCH}.lib")
+   else()
+       file(GLOB_RECURSE SOCI_LIBRARIES "${SOCI_LIBRARY_DIR}/Release/*.lib")
+       list(APPEND SOCI_LIBRARIES "${WIN_LDAP_LIBRARY}")
+       list(APPEND SOCI_LIBRARIES "${WIN_SSPI_LIBRARY}")
+       list(APPEND SOCI_LIBRARIES "${RSTUDIO_WINDOWS_DEPENDENCIES_DIR}/install-soci/sqlite/sqlite3-release-${SOCI_ARCH}.lib")
+       list(APPEND SOCI_LIBRARIES "${RSTUDIO_WINDOWS_DEPENDENCIES_DIR}/install-soci/postgresql/lib/libpq-release-${SOCI_ARCH}.lib")
+       list(APPEND SOCI_LIBRARIES "${RSTUDIO_WINDOWS_DEPENDENCIES_DIR}/install-soci/postgresql/lib/libpgcommon-release-${SOCI_ARCH}.lib")
+       list(APPEND SOCI_LIBRARIES "${RSTUDIO_WINDOWS_DEPENDENCIES_DIR}/install-soci/postgresql/lib/libpgport-release-${SOCI_ARCH}.lib")
+   endif()
+endif()
+list(LENGTH SOCI_LIBRARIES SOCI_LIB_COUNT)
+
+if (SOCI_LIB_COUNT EQUAL 0)
+   message(FATAL_ERROR "No SOCI libraries found under ${SOCI_LIBRARY_DIR}. Ensure the SOCI dependency is installed and try again.")
+else()
```

Comment:
Nit: because the above is a fatal error you could just `endif()` and save a level of indentation.